test/resolve/adapters/decomposedSourceAdapter.ts

```diff
@@ -39,6 +36,27 @@ describe('DecomposedSourceAdapter', () => {
     );
   });
 
+  it('should return parent SourceComponent when given a child xml of non-addressable type', () => {
+    const nonAddressableType = JSON.parse(
+      JSON.stringify(mockRegistryData.types.decomposedtoplevel)
```

Comment:
why `parse` and `stringify` shouldn't those cancel each other out?

Reply:
to make a deep clone of the object so I can modify it and not affect other tests.